src/set_theory/game.lean

```diff
@@ -0,0 +1,1221 @@
+begin
+  rw [le_def],
+  split,
+  { intro i,
+    change xr ⊕ xl at i,
+    cases i,
+    { -- If Left played in -x, Right responds with the same move in x.
+      right,
+      fsplit,
+      { dsimp,
+        apply right_moves_add.inv_fun,
+        right,
+        use i },
+      { dsimp,
+        simp only [move_right_mk, add_move_right_inr],
+        apply add_left_neg_le_zero, } },
+    { -- If Left in x, Right responds with the same move in -x.
+      right,
+      fsplit,
+      { dsimp,
+        apply right_moves_add.inv_fun,
+        left,
+        use i },
+      { dsimp,
+        simp only [move_right_mk, add_move_right_inl],
+        apply add_left_neg_le_zero, } },
+  },
+  { rintro ⟨⟩, }
+end
+using_well_founded { dec_tac := pgame_wf_tac }
+
+theorem zero_le_add_left_neg : Π {x : pgame}, 0 ≤ (-x) + x
+| ⟨xl, xr, xL, xR⟩ :=
+begin
+  rw [le_def],
+  split,
+  { rintro ⟨⟩, },
+  { intro i,
+    change xl ⊕ xr at i,
+    cases i,
+    { -- If Right played in -x, Left responds with the same move in x.
+      left,
+      fsplit,
+      { dsimp,
+        apply left_moves_add.inv_fun,
+        right,
+        use i },
+      { dsimp,
+        simp only [move_left_mk, add_move_left_inr],
+        apply zero_le_add_left_neg, } },
+    { -- If Right in x, Left responds with the same move in -x.
+      left,
+      fsplit,
+      { dsimp,
+        apply left_moves_add.inv_fun,
+        left,
+        use i },
+      { dsimp,
+        simp only [move_left_mk, add_move_left_inl],
+        apply zero_le_add_left_neg, } },
+  },
+end
+using_well_founded { dec_tac := pgame_wf_tac }
+
+theorem add_left_neg_equiv {x : pgame} : (-x) + x ≈ 0 :=
+⟨add_left_neg_le_zero, zero_le_add_left_neg⟩
+
+theorem add_right_neg_le_zero {x : pgame} : x + (-x) ≤ 0 :=
+calc x + (-x) ≤ (-x) + x : le_of_relabelling (add_comm_relabelling _ _)
+     ... ≤ 0 : add_left_neg_le_zero
+
+theorem zero_le_add_right_neg {x : pgame} : 0 ≤ x + (-x) :=
+calc 0 ≤ (-x) + x : zero_le_add_left_neg
+     ... ≤ x + (-x) : le_of_relabelling (add_comm_relabelling _ _)
+
+theorem add_lt_add_right {x y z : pgame} (h : x < y) : x + z < y + z :=
+begin
+  rw ←not_le,
+  rw ←not_le at h,
+  intro w,
+  replace w : (y + z) + (-z) ≤ (x + z) + (-z) := by apply add_le_add_right w,
+  have h' : y ≤ x,
+  calc y ≤ y + 0            : le_of_relabelling (add_zero_relabelling _).symm
+       ... ≤ y + (z + -z)   : @add_le_add_left y 0 (z + -z) zero_le_add_right_neg
+       ... ≤ (y + z) + (-z) : le_of_relabelling (add_assoc_relabelling _ _ _).symm
+       ... ≤ (x + z) + (-z) : w
+       ... ≤ x + (z + -z)   : le_of_relabelling (add_assoc_relabelling _ _ _)
+       ... ≤ x + 0          : @add_le_add_left x (z + -z) 0 add_right_neg_le_zero
+       ... ≤ x              : le_of_relabelling (add_zero_relabelling _),
+  exact h h',
+end
+
+theorem add_lt_add_left {x y z : pgame} (h : y < z) : x + y < x + z :=
+-- We can't argue via `x + y ≤ y + x < z + x ≤ x + z` since pgame does not form a preorder.
```

Comment:
Aren't `lt_of_le_of_lt` and `lt_of_lt_of_le` nevertheless true, though? Or am I confused again?